code/components/extra-natives-five/src/RuntimeAssetNatives.cpp

```diff
@@ -897,6 +916,22 @@ fwArchetype* GetArchetypeSafe(uint32_t archetypeHash, uint64_t* archetypeUnk)
 
 static InitFunction initFunction([]()
 {
+	OnMainGameFrame.Connect([]
+	{
+		decltype(mainFrameQueue) q;
```

Comment:
True, I did read that some non-x86 architectures may not update such values atomically, so I always take such race conditions into account. But the rest of the code is already thread-safe so that would've resolved it anyway.